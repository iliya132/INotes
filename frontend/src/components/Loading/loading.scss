.loading-container {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 90%;
}

p {
    margin-right: 15px;
}
