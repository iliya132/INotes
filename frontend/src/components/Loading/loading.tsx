import React from 'react';
import ReactLoading from 'react-loading';
import styles from './loading.scss';

export function Loading() {
    return (
        <div className={styles["loading-container"]}>
            <p>Загрузка. Подождите...</p>
            <ReactLoading type="cylon" color="black" />
        </div>
    )
}
