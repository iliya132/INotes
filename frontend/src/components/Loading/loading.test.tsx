/* eslint-disable no-undef */
import React from 'react';
import { createRoot, Root } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import { Loading } from './loading';

//#region common setup
let root: Root | null = null;
let container: Element | null = null;
beforeEach(() => {
    act(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        root = createRoot(container);
    });
});

afterEach(() => {
    act(() => {
        root!.unmount();
        container!.remove();
        container = null;
    });
});
//#endregion

describe("test  input component", () => {
    it("matches snapshot", () => {
        act(() => {
            root!.render(<Loading />)
        })
        expect(container!.firstChild).toMatchSnapshot("Loading test snapshot")
    })
})