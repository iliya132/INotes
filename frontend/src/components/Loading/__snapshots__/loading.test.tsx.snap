// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  input component matches snapshot: Loading test snapshot 1`] = `
<div
  class="loading-container"
>
  <p>
    Загрузка. Подождите...
  </p>
  <div
    style="fill: black; height: 64px; width: 64px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M0 12 V20 H4 V12z"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.5"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.1s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.25"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.2s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</div>
`;
